### ISSUE TITLE
New message creation notification fails

### ISSUE THREAD

**Issue by verement** [humhub/humhub-modules-mail#173]
When a new message is created, email notification to the recipient fails. The only indication is this log message:

```
Could not send notification e-mail to: username. Error:Trying to get property 'content' of non-object
```

This message originates from `protected/modules/mail/models/forms/CreateMessage.php` line 144. By removing the try/catch wrapper, the following stack trace is revealed:

```
#0 /var/www/humhub/protected/modules/mail/views/emails/NewMessage.php(124): yii\base\ErrorHandler->handleError(8, 'Trying to get p...', '/var/www/humhub...', 124, Array)
#1 /var/www/humhub/protected/vendor/yiisoft/yii2/base/View.php(348): require('/var/www/humhub...')
#2 /var/www/humhub/protected/vendor/yiisoft/yii2/base/View.php(257): yii\base\View->renderPhpFile('/var/www/humhub...', Array)
#3 /var/www/humhub/protected/vendor/yiisoft/yii2/base/View.php(156): yii\base\View->renderFile('/var/www/humhub...', Array, Object(humhub\components\mail\Mailer))
#4 /var/www/humhub/protected/vendor/yiisoft/yii2/mail/BaseMailer.php(302): yii\base\View->render('@mailmodule/vie...', Array, Object(humhub\components\mail\Mailer))
#5 /var/www/humhub/protected/vendor/yiisoft/yii2/mail/BaseMailer.php(188): yii\mail\BaseMailer->render('@mailmodule/vie...', Array, 'layouts/html')
#6 /var/www/humhub/protected/humhub/components/mail/Mailer.php(78): yii\mail\BaseMailer->compose(Array, Array)
#7 /var/www/humhub/protected/modules/mail/models/Message.php(239): humhub\components\mail\Mailer->compose(Array, Array)
#8 /var/www/humhub/protected/modules/mail/models/forms/CreateMessage.php(142): humhub\modules\mail\models\Message->notify(Object(humhub\modules\user\models\User))
#9 /var/www/humhub/protected/modules/mail/models/forms/CreateMessage.php(100): humhub\modules\mail\models\forms\CreateMessage->saveRecipients()
#10 /var/www/humhub/protected/modules/mail/controllers/MailController.php(308): humhub\modules\mail\models\forms\CreateMessage->save()
#11 [internal function]: humhub\modules\mail\controllers\MailController->actionCreate()
#12 /var/www/humhub/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#13 /var/www/humhub/protected/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#14 /var/www/humhub/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\base\Controller->runAction('create', Array)
#15 /var/www/humhub/protected/vendor/yiisoft/yii2/web/Application.php(103): yii$base\Module->runAction('mail/mail/creat...', Array)
#16 /var/www/humhub/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\web\Application->handleRequest(Object(humhub\components\Request))
#17 /var/www/humhub/index.php(25): yii\base\Application->run()
#18 {main}
```

See also [this post](https://community.humhub.com/content/perma?id=211754).

**Comment by R3callGames** [comment]
same issue on the latest version, no matter what transport (file, php, or smtp)

**Comment by verement** [comment]
After a little further digging I believe this problem is caused by commit e9da798f1560b7906a7bd358c9d370aee92cbb42.

When `saveRecipients()` is called before `saveMessageEntry()`, the message entry is not available to be retrieved by `Message::getLastEntry()` which is used when populating the notification `$entry` in `Message::notify()` (`protected/modules/mail/models/Message.php` line 238).

**Comment by R3callGames** [comment]
overwrote that file with the previous message and it's working as expected

**Comment by luke-** [comment]
Fixed in 1.0.15

This change was required for an individual customization.
Thanks for the report and information
